### PR TITLE
tpm_device: add invalid xml test for external vtpm

### PR DESCRIPTION
**Pull request by yanqzhan**
Add invalid xml test for external vtpm, including nonexist, empty and invalid values for source_type, source_mode and source_socket.
To auto VIRT-296945.

### CODE REVIEW

**Comment by yanqzhan** [comment]
Local run pass:
avocado run --vt-type libvirt --vt-machine-type q35 --test-runner=runner tpm_device.negative_test.external
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 0fe8a68398705701a4f77c6bb0513e467f34d8a6
JOB LOG    : /var/lib/avocado/job-results/job-2023-05-08T04.56-0fe8a68/job.log
 (1/8) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.negative_test.external.source_type.nonexist: PASS (5.10 s)
 (2/8) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.negative_test.external.source_type.empty: PASS (5.16 s)
 (3/8) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.negative_test.external.source_type.invalid: PASS (5.13 s)
 (4/8) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.negative_test.external.source_mode.nonexist: PASS (5.26 s)
 (5/8) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.negative_test.external.source_mode.empty: PASS (5.13 s)
 (6/8) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.negative_test.external.source_mode.invalid: PASS (5.14 s)
 (7/8) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.negative_test.external.source_path.nonexist: PASS (5.16 s)
 (8/8) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.negative_test.external.source_path.empty: PASS (5.49 s)
RESULTS    : PASS 8 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-05-08T04.56-0fe8a68/results.html
JOB TIME   : 42.98 s

File: libvirt/tests/cfg/virtual_device/tpm_device.cfg [hunk 1]
```diff
@@ -242,3 +242,40 @@
...
+                    func_supported_since_libvirt_ver = (9, 0, 0)
+                    backend_type = 'external'
+                    no pseries, s390-virtio
+                    q35:
+                        tpm_model = 'tpm-crb'
+                    aarch64:
+                        tpm_model = 'tpm-tis'
+                    variants:
+                        - source_type:
+                            xml_errmsg = "XML error: missing external TPM backend source type"
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virtual_device/tpm_device.cfg:255]: xml_errms --> xml_error_msg

File: libvirt/tests/cfg/virtual_device/tpm_device.cfg [hunk 2]
```diff
@@ -242,3 +242,40 @@
...
+                            variants:
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virtual_device/tpm_device.cfg:256]: maybe this can be redesign as below to avoid duplicated nonexist, empty, invalid:
```
varianst:
     - source_code
     -  source_mode:
  varinasts:
       - nonexist
       - empty
       - invalid
     ```

**Review by chloerh — APPROVED** [review]

File: libvirt/tests/cfg/virtual_device/tpm_device.cfg (hunk 2 above)
Comment (yanqzhan) [reply libvirt/tests/cfg/virtual_device/tpm_device.cfg:256]: They have different error msg and attr config for each scenario, is it more complex if change to this?
                    variants:
                        - source_type:
                            config_attr = source_type
                            xml_errmsg = "XML error: missing external TPM backend source type"
                            invalid:
                                xml_errmsg = "unsupported configuration: unknown backend source type 'unixaa' for external TPM"
                        - source_mode:
                            config_attr = source_mode
                            nonexist:
                                status_error = 'no'
                            empty:
                                xml_errmsg = "XML error: Invalid value for attribute 'mode' in element 'source': ''"
                            invalid：
                                xml_errmsg = "XML error: only 'connect' mode is supported for external TPM device"
                        - source_path:
                            config_attr = source_path
                            nonexist:
                                xml_errmsg = "error: XML error: missing socket path for external TPM device"
                            empty:
                                status_error = 'no'
                            no invalid
                    variants:
                        - nonexist:
                            config_behavior = 'nonexist'
                        - empty:
                            config_behavior = 'empty'
                        - invalid
                            config_behavior = 'invalid'

File: libvirt/tests/cfg/virtual_device/tpm_device.cfg (hunk 1 above)
Comment (yanqzhan) [reply libvirt/tests/cfg/virtual_device/tpm_device.cfg:255]: All my previous contents in .cfg are using xml_errmsg, could I keep it? 
Is there any principle that we must use xml_error_msg please?

Comment (chunfuwen) [reply libvirt/tests/cfg/virtual_device/tpm_device.cfg:255]: it is just python naming convention, separate word is connected with "_".
But it is trivial things

File: libvirt/tests/cfg/virtual_device/tpm_device.cfg (hunk 2 above)
Comment (chunfuwen) [reply libvirt/tests/cfg/virtual_device/tpm_device.cfg:256]: OK, please keep it currently

**Review by chunfuwen — APPROVED** [review]
lgtm

**Comment by yanqzhan** [comment]
Thank you!

File: libvirt/tests/cfg/virtual_device/tpm_device.cfg (hunk 1 above)
Comment (yanqzhan) [reply libvirt/tests/cfg/virtual_device/tpm_device.cfg:255]: Get it. Thanks!
